Assets/Scripts/Player/GridCombatSystem.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using OperationBlackwell.Core;
+
+namespace OperationBlackwell.Player {
+	public class GridCombatSystem : MonoBehaviour {
+		[SerializeField] private UnitGridCombat[] unitGridCombatArray_;
+
+		private State state_;
+		private UnitGridCombat unitGridCombat_;
+		private List<UnitGridCombat> blueTeamList_;
+		private List<UnitGridCombat> redTeamList_;
+		private int blueTeamActiveUnitIndex_;
+		private int redTeamActiveUnitIndex_;
+
+		private enum State {
+			Normal,
+			Waiting
+		}
+
+		private void Awake() {
+			state_ = State.Normal;
+		}
+
+		private void Start() {
+			blueTeamList_ = new List<UnitGridCombat>();
+			redTeamList_ = new List<UnitGridCombat>();
+			blueTeamActiveUnitIndex_ = -1;
+			redTeamActiveUnitIndex_ = -1;
+
+			// Set all UnitGridCombat on their GridPosition
+			foreach (UnitGridCombat unitGridCombat_ in unitGridCombatArray_) {

```

Comment:
Coding style: Don't add a space after your foreach-loop and if-statements in thie file